python/ray/rllib/optimizers/aso_tree_aggregator.py

```diff
@@ -0,0 +1,156 @@
+"""Helper class for AsyncSamplesOptimizer."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import logging
+import os
+import time
+
+import ray
+from ray.rllib.utils.actors import TaskPool, create_colocated
+from ray.rllib.utils.annotations import override
+from ray.rllib.optimizers.aso_aggregator import Aggregator, \
+    AggregationWorkerBase
+
+logger = logging.getLogger(__name__)
+
+
+class TreeAggregator(Aggregator):
+    """A hierarchical experiences aggregator.
+
+    The given set of remote evaluators is divided into subsets and assigned to
+    one of several aggregation workers. These aggregation workers collate
+    experiences into batches of size `train_batch_size` and we collect them
+    in this class when `iter_train_batches` is called.
+    """
+
+    def __init__(self,
```

Comment:
I don't think this is helpful. These attributes are already documented elsewhere, so documenting them again would just cause the comment to be repeated, and hence easily get out of date / stale.

Reply:
Yeah I see - I think at least referencing where they are documented would help; I only bring this up because that was one pain point in personally reviewing this code